test/vec_notation.lean

```diff
@@ -0,0 +1,24 @@
+import Mathlib.Data.Fin.VecNotation
+
+/-! These tests are testing `PiFin.toExpr` and fail with
+`local attribute [-instance] PiFin.toExpr` -/
+
+open Lean.Meta
+
+#eval do
+  let x : Fin 0 → ℕ := ![]
+  isDefEq `(x) `(![] : Fin 0 → ℕ)
```

Comment:
These tests are hand-ported from mathlib3, but I don't know what the new syntax is for `` `(x) ``